cjs/html/element.js

```diff
@@ -32,8 +32,15 @@ class HTMLElement extends Element {
 
   constructor(ownerDocument = null, localName = '') {
     super(ownerDocument, localName);
+    const {constructor: Class, [END]: end} = this;
+    if (upgradingElements.has(Class)) {
```

Comment:
this has a performance impact for everything ... from zero `.has` checks, we now pass through a WeakMap for every created element. I don't like it. This whole project degrades performance on demand: you use MutationObserver? One level perf degradation (inevitable) ... you use the Custom Elements Registry? Another level of perf degradation.

Accordingly, I would probably consider something like this instead:

```js

  constructor(ownerDocument = null, localName = '') {
    super(ownerDocument, localName);
    const {constructor: Class, [END]: end} = this;
    if (!ownerDocument) {
      if (!Classes.has(Class))
        throw new Error('unable to initialize this Custom Element');
      const {ownerDocument, localName, options} = Classes.get(Class);
      this.ownerDocument = end.ownerDocument = ownerDocument;
      this.localName = localName;
      customElements.set(this, {connected: false});
      if (options.is)
        this.setAttribute('is', options.is);
    }
    else {
      const {[CUSTOM_ELEMENTS]: {active}} = ownerDocument;
      if (active && upgradingElements.has(Class)) {
        // ... that thing
      }
    }
  }
```

However, I am not even sure if that's faster or slower, but that's how it's done for MutationObserver too so it's surely more consistent.